jupyter_sphinx/execute.py

```diff
@@ -185,6 +187,21 @@ def run(self):
             self.assert_has_content()
             content = self.content
 
+        emphasize_linespec = self.options.get('emphasize-lines')
+        if emphasize_linespec:
+            try:
+                nlines = len(content)
+                hl_lines = parselinenos(emphasize_linespec, nlines)
+                if any(i >= nlines for i in hl_lines):
+                    logger.warning(
+                        'Line number spec is out of range(1-{}): {}'.format(
+                            nlines, emphasize_linespec), location=location)
+                hl_lines = [i + 1 for i in hl_lines if i < nlines]
+            except ValueError as err:
```

Comment:
I don't quite understand which error we're catching here, can you explain?

Or is this implementation a direct copy of what's in code-block?

Reply:
> Or is this implementation a direct copy of what's in code-block?

It is about [the same](https://github.com/sphinx-doc/sphinx/blob/0319faf8f1503453b6ce19020819a8cf44e39f13/sphinx/directives/code.py#L134-L148). :)

We are catching ValueError from `parselinenos` function `invalid line number spec`.

Reply:
Thanks, can you please refer to the original code and add this comment inline?

Other than that, everything is good to go as far as I can see.

Reply:
Done